app/assets/javascripts/browsered/web-payments/index.js

```diff
@@ -1,36 +1,46 @@
 'use strict'
 
-const { prepareRequestObject, clearErrorSummary } = require('./helpers')
+const { clearErrorSummary } = require('./helpers')
 const makeApplePayRequest = require('./apple-pay')
-const makePaymentRequest = require('./google-pay')
+const { createGooglePaymentRequest, googlePayNow } = require('./google-pay')
 
 module.exports = () => {
   const paymentMethodForm = document.getElementById('payment-request-container')
   const standardMethodContainer = document.getElementById('enter-card-details-container')
 
   if (window.PaymentRequest && paymentMethodForm) {
-    paymentMethodForm.classList.remove('hidden')
-    standardMethodContainer.classList.add('hidden')
-
     if (window.ApplePaySession && ApplePaySession.canMakePayments()) {
-      document.getElementById('payment-method-payment-request-apple').parentNode.style.display = 'block'
-    } else if (/Chrome/.test(navigator.userAgent) && /Google Inc/.test) {
-      document.getElementById('payment-method-payment-request').parentNode.style.display = 'block'
+      paymentMethodForm.classList.remove('hidden')
+      standardMethodContainer.classList.add('hidden')
+      document.getElementById('payment-method-apple-pay').parentNode.style.display = 'block'
+      document.getElementById('payment-method-apple-pay').checked = true
+    } else {
+      console.log('payment request API is available')
+      const request = createGooglePaymentRequest();
+      request.canMakePayment()
+        .then(result => {
+          if (result) {
+            console.log('Google Pay is available')
+            paymentMethodForm.classList.remove('hidden')
+            standardMethodContainer.classList.add('hidden')
+            document.getElementById('payment-method-google-pay').parentNode.style.display = 'block'
+          }
+        })
```

Comment:
i take it all the above is saying the user can pay with either Apple Pay or G Pay but not both?

Reply:
Yeah so we know webpayments are available on the service but then we check the browser is enabled to do them, then we trigger Apple Pay if it’s Safari or else Google Pay